apps/cowswap-frontend/src/modules/application/containers/TradeWidgetLinks/index.tsx

```diff
@@ -50,13 +59,20 @@ export function TradeWidgetLinks({
   const tradeContext = useTradeRouteContext()
   const location = useLocation()
   const [isDropdownVisible, setDropdownVisible] = useState(false)
+  const { enabledTradeTypes } = useInjectedWidgetParams()
 
   const handleMenuItemClick = (_item?: MenuItemConfig) => {
     if (menuItems.length === 1) return
     setDropdownVisible(false)
   }
 
-  const menuItems = MENU_ITEMS.map((item) => {
+  const enabledItems = MENU_ITEMS.filter((item) => {
+    if (!enabledTradeTypes?.length) return true
```

Comment:
not a big deal since they are 3 elements, but i thnk is easier to check the length out of the loop and not even do the loop if there's not `enabledTradeTypes`